src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -20,25 +22,37 @@ protected WebKitWebViewClient(IntPtr javaReference, JniHandleOwnership transfer)
 		{
 			// This constructor is called whenever the .NET proxy was disposed, and it was recreated by Java. It also
 			// happens when overridden methods are called between execution of this constructor and the one above.
-			// because of these facts, we have to check
-			// all methods below for null field references and properties.
+			// because of these facts, we have to check all methods below for null field references and properties.
 		}
 
 		public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceRequest? request)
 		{
-			// handle redirects to the app custom scheme by reloading the url in the view.
-			// otherwise they will be blocked by Android.
+			// Handle redirects to the app custom scheme by reloading the URL in the view.
+			// Handle navigation to external URLs using the system browser, unless overriden.
 			var requestUri = request?.Url?.ToString();
-			if (requestUri != null && view != null &&
-				request != null && request.IsRedirect && request.IsForMainFrame)
+			if (requestUri != null)
 			{
 				var uri = new Uri(requestUri);
-				if (uri.Host == "0.0.0.0")
+
+				if (uri.Host == "0.0.0.0" &&
```

Comment:
I realize this code was already here, but maybe we can store the string `0.0.0.0` in some field and use it everywhere? Example: `private const string AppHostAddress = "0.0.0.0";`